lib/opal/errors.rb

```diff
@@ -12,4 +16,17 @@ def initialize(gem_name)
       super("can't find gem #{gem_name}")
     end
   end
+
+  class CompilationError < Error
+    attr_accessor :location
```

Comment:
I think we could add a constructor for this attribute and use it with a `raise CompilationError, msg, location` or `raise CompilationError, msg, location: location`.

Reply:
> The optional second parameter sets the message associated with the exception, and the third parameter is an array of callback information. 

Can't be done AFAIK

Reply:
Let's keep it as is then